System/OsPath/Data/ByteString/Short/Word16.hs

```diff
@@ -647,6 +650,22 @@ splitWith p = \(assertEven -> sbs) -> if
               | otherwise -> a : go (tail b)
 
 
+-- | Check whether one string is a substring of another.
+isInfixOf :: ShortByteString -> ShortByteString -> Bool
+isInfixOf sbs = \s -> null sbs || not (null $ snd $ GHC.Exts.inline breakSubstring sbs s)
+
+
+breakSubstring :: ShortByteString -- ^ String to search for
+               -> ShortByteString -- ^ String to search in
+               -> (ShortByteString, ShortByteString) -- ^ Head and tail of string broken at substring
+breakSubstring pat = go mempty
+ where
+    go prefix s = let (h, t)  = GHC.Exts.inline BS.breakSubstring pat s
+                  in if | even (BS.length h) -> (prefix `append` h, t)
+                        | otherwise -> let (p, s') = BS.splitAt (BS.length h + 1) s
+                                       in go (prefix `append` p) s'
```

Comment:
> you can probably rely on the fact that `B.breakSubstring` return slices of the original string

Is this indeed the case? Can a `ShortByteString` be sliced?

Reply:
OK, so I think the root of the confusion is that @hasufell linked to `Data.ByteString` instead of `Data.ByteString.Short` at https://github.com/haskell/filepath/issues/195#issuecomment-1604069469.

@Bodigrim @hasufell is that the case?

Reply:
ShortByteString makes copies.